tests/test_stability_funs.py

```diff
@@ -349,3 +351,403 @@ def test_magwell_print(capsys):
+    ntor = 2
+
+    eq0 = desc.examples.get("W7-X")
+    eq1 = desc.examples.get("precise_QA")
+
+    eq_list = [eq0, eq1]
+
+    for eq in eq_list:
+        eq_keys = ["iota", "iota_r", "a", "rho", "psi"]
+
+        grid = QuadratureGrid(eq.L_grid, eq.M_grid, eq.N_grid, NFP=eq.NFP)
+        data_eq = eq.compute(eq_keys, grid=grid)
+
+        fi = interp1d(
+            data_eq["rho"][grid.unique_rho_idx],
+            data_eq["iota"][grid.unique_rho_idx],
+        )
+        fs = interp1d(
+            data_eq["rho"][grid.unique_rho_idx],
+            data_eq["iota_r"][grid.unique_rho_idx],
+        )
+
+        iotas = fi(np.sqrt(psi))
+        shears = fs(np.sqrt(psi))
+
+        rho = np.sqrt(psi)
+        N = (8 * eq.M_grid * eq.N_grid) * ntor + 1
+        zeta = np.linspace(-ntor * np.pi, ntor * np.pi, N)
+
+        data_keys = [
+            "p_r",
+            "psi",
+            "psi_r",
+            "sqrt(g)_PEST",
+            "|grad(psi)|^2",
+            "grad(|B|)",
+            "grad(alpha)",
+            "grad(psi)",
+            "B",
+            "grad(|B|)",
+            "kappa",
+            "iota",
+            "lambda_t",
+            "lambda_z",
+            "lambda_tt",
+            "lambda_zz",
+            "lambda_tz",
+            "g^aa",
+            "g^ra",
+            "g^rr",
+            "cvdrift",
+            "cvdrift0",
+            "|B|",
+            "B^zeta",
+        ]
+
+        grid = Grid.create_meshgrid(
+            [rho, alpha, zeta],
+            coordinates="raz",
+            period=(np.inf, 2 * np.pi, np.inf),
+        )
+
+        data = eq.compute(data_keys, grid=grid)
+
+        psi_s = data_eq["psi"][-1]
+        sign_psi = psi_s / np.abs(psi_s)
+        sign_iota = iotas / np.abs(iotas)
+        # normalizations
+        Lref = data_eq["a"]
+        Bref = 2 * np.abs(psi) / Lref**2
+
+        modB = data["|B|"]
+        x = Lref * np.sqrt(psi)
+        shat = -x / iotas * shears / Lref
+
+        psi_r = data["psi_r"]
+
+        grad_psi = data["grad(psi)"]
+        grad_psi_sq = data["|grad(psi)|^2"]
+        grad_alpha = data["grad(alpha)"]
+
+        g_sup_rr = data["g^rr"]
+        g_sup_ra = data["g^ra"]
+        g_sup_aa = data["g^aa"]
+
+        modB = data["|B|"]
+
+        B_sup_zeta = data["B^zeta"]
+
+        gds2 = np.array(dot(grad_alpha, grad_alpha)) * Lref**2 * psi
+        gds2_alt = g_sup_aa * Lref**2 * psi
+
+        gds21 = -sign_iota * np.array(dot(grad_psi, grad_alpha)) * shat / Bref
+        gds21_alt = -sign_iota * g_sup_ra * shat / Bref * (psi_r)
+
+        gds22 = grad_psi_sq * (1 / psi) * (shat / (Lref * Bref)) ** 2
+        gds22_alt = g_sup_rr * (psi_r) ** 2 * (1 / psi) * (shat / (Lref * Bref)) ** 2
+
+        gbdrift = np.array(dot(cross(data["B"], data["grad(|B|)"]), grad_alpha))
+        gbdrift *= -sign_psi * 2 * Bref * Lref**2 / modB**3 * np.sqrt(psi)
+        gbdrift_alt = -sign_psi * data["gbdrift"] * 2 * Bref * Lref**2 * np.sqrt(psi)
+
+        cvdrift = (
```

Comment:
in the compute quantity cvdrift, we add gbdrift to 1/B^2 * dp/dpsi. What is this alternate way of computing it?

Reply:
b cross kappa dot grad(alpha). These are two different ways and they only match if an equilibrium is in good force balance.